json/src/lib.rs

```diff
@@ -1812,6 +1812,84 @@ pub struct FundRawTransactionResult {
     pub change_position: i32,
 }
 
+#[derive(Clone, PartialEq, Eq, Debug, Default)]
+pub struct BumpFeeOptions {
+    pub conf_target: Option<u16>,
+    /// Specify a fee rate instead of relying on the built-in fee estimator.
+    pub fee_rate: Option<FeeRate>,
+    /// Whether this transaction could be replaced due to BIP125 (replace-by-fee)
+    pub replaceable: Option<bool>,
+    /// The fee estimate mode
+    pub estimate_mode: Option<EstimateMode>,
+}
+
+impl BumpFeeOptions {
+    pub fn to_serializable(&self, version: usize) -> SerializableBumpFeeOptions {
+        let fee_rate = self.fee_rate.map(|x| {
+            if version < 210000 {
+                x.btc_per_kvbyte()
+            } else {
+                x.sat_per_vbyte()
+            }
+        });
+
+        SerializableBumpFeeOptions {
+            fee_rate,
+            conf_target: self.conf_target,
+            replaceable: self.replaceable,
+            estimate_mode: self.estimate_mode,
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Default)]
+pub struct FeeRate(Amount);
```

Comment:
Hmm this seems a useful utility type. Could you perhaps put it somewhere more at the top?

Also, I think it should have some constructors like `per_vbyte(Amount) -> FeeRate` and `per_vkb(Amount) -> FeeRate`.

And then the two conversion methods could start with `as_` not to be confused with the constructors. If Core actually made the change from btc/vkb to sat/vb then we might use this type more in the future.

Reply:
Ah could you also document the type?